task/source-build/0.1/source-build.yaml

```diff
@@ -0,0 +1,472 @@
+                return False
+
+            # bsi reads source RPMs from this directory
+            bsi_rpms_dir = create_dir(base_image_sources_dir, "bsi_rpms_dir")
+            sib_dirs.rpm_dir = str(bsi_rpms_dir)  # save this directory for executing bsi
+
+            # bsi reads extra sources from this directory
+            # each source is in its own directory, for instance, subdir/source_a.tar.gz
+            extra_src_dir = create_dir(base_sources_extraction_dir, "extra_src_dir")
+
+            # extract layers, primarily they are RPMs
+            with open(f"{base_sources_extraction_dir}/manifest.json", "r") as f:
+                manifest_data = json.load(f)
+
+            for layer in manifest_data["layers"]:
+                digest = layer["digest"].split(":")[-1]
+                log.debug("untar layer %s", digest)
+
+                blob_member = ""
+                symlink_member = ""
+                with tarfile.open(f"{base_sources_extraction_dir}/{digest}", "r:gz") as tar:
+                    for member in tar:
+                        if member.isfile():
+                            blob_member = member.name
+                        elif member.issym():
+                            symlink_member = member.name
+
+                if symlink_member.startswith("./rpm_dir/"):
+                    dest_dir = sib_dirs.rpm_dir
+                    log.debug("Prepare SRPM %s", symlink_member)
+
+                    extract_blob_member(digest, blob_member, dest_dir,
+                                        rename_to=os.path.basename(symlink_member),
+                                        work_dir=base_sources_extraction_dir,
+                                        log=log)
+
+                elif symlink_member.startswith("./extra_src_dir/"):
+                    extra_src_archive = os.path.basename(symlink_member)
+                    log.debug("Prepare extra source %s", extra_src_archive)
+                    # one extra source archive per directory, no matter what the directory name is.
+                    dest_dir = create_dir(extra_src_dir, extra_src_archive)
+                    sib_dirs.extra_src_dirs.append(dest_dir)  # save this directory for bsi
+
+                    extract_blob_member(digest, blob_member, str(dest_dir),
+                                        rename_to=extra_src_archive,
+                                        work_dir=base_sources_extraction_dir,
+                                        log=log)
+
+                    # FIXME: perhaps the dependency archive (the Cachi2) might be handled differently
+                    run(["tar", "xvf", extra_src_archive], check=True, cwd=dest_dir)
+                    os.unlink(f"{dest_dir}/{extra_src_archive}")
+
+                else:
+                    log.warning("No known operation happened on layer %s", digest)
+
+            return len(os.listdir(sib_dirs.rpm_dir)) > 0 or len(sib_dirs.extra_src_dirs) > 0
```

Comment:
sib_dirs.extra_src_dirs will most likely already have 1 element with sources from make_source_archive

so it should be len(sib_dirs.extra_src_dirs) > 1

Reply:
Good catch.

Reply:
Updated. Now, using a flag variable instead.